Scripts/Runtime/Entities/Lifecycle/EntitySpawnSystem.cs

```diff
@@ -57,10 +67,74 @@ protected override void OnDestroy()
         {
             m_ActiveEntitySpawners.Clear();
             m_EntityArchetypes.Dispose();
+            m_EntityPrototypes.Dispose();
             m_EntitySpawners.DisposeAllValuesAndClear();
             base.OnDestroy();
         }
 
+        private void CreateArchetypeLookup()
+        {
+            foreach ((Type definitionType, IEntitySpawnDefinition entitySpawnDefinition) in EntitySpawnSystemReflectionHelper.SPAWN_DEFINITION_TYPES)
+            {
+                long entityArchetypeHash = BurstRuntime.GetHashCode64(definitionType);
+                if (m_EntityArchetypes.TryGetValue(entityArchetypeHash, out EntityArchetype entityArchetype))
+                {
+                    continue;
+                }
+
+                if (definitionType.GetCustomAttribute<BurstCompatibleAttribute>() == null)
```

Comment:
I don't think so because you won't annotate your `IEntitySpawnDefinition` structs with `BurstCompile`, only `BurstCompatible` 

The jobs that use it will Burst the functions they need.

Reply:
Ah right, good point